internal/slicer/report.go

```diff
@@ -0,0 +1,61 @@
+package slicer
+
+import (
+	"fmt"
+	"github.com/canonical/chisel/internal/fsutil"
+	"github.com/canonical/chisel/internal/setup"
+	"io/fs"
+	"sync"
+)
+
+type FileReport struct {
+	Path    string
+	Mode    fs.FileMode
+	Hash    string
+	Size    uint
+	Mutable bool
+	Slices  map[*setup.Slice]bool
+	Link    string
+}
```

Comment:
What do you think about keeping one type -- `fsutil.FileInfo`? Or, `slicer.Report` extend the `fsutil.FileInfo` type?

Reply:
By extend you mean embed the other struct?

I want to keep them separate because `fsutil.FileInfo` has no notion of slices nor of mutability and adding the fields there would break the abstraction a little. At the same time, I dont like embedding that much because it ties the way we represent fields in the proxy with how we represent them here.

I am curious to know more about your suggestion, can you add more information?

Reply:
Okay. Reason I asked was the similarity between two structs.